client/src/contexts/AuthenticationContext/AuthenticationProvider.tsx

```diff
@@ -117,7 +119,10 @@ const AuthenticationProvider = (props: PropsWithChildren<IAuthenticationProvider
         keycloak.clearToken()
 
         return setAuthenticationTokens(undefined)
-      } else if (decodedAccessToken?.exp && decodedAccessToken.exp <= Date.now() / 1000 + refreshTokenFrequency) {
+      } else if (
+        decodedAccessToken?.exp &&
+        decodedAccessToken.exp <= Date.now() / 1000 + refreshTokenFrequency
```

Comment:
The idea is that the token is already refreshed when it would expire before the next check. I check every refreshTokenFrequency whether the token is expired and it just adds this interval to the check.
I need to divide Date.now() by 1000 because it returns milliseconds and in the JWT token the timestamp is stored in seconds